### ISSUE TITLE
[feat] Ability to force a node to a dead/failed state

### ISSUE THREAD

**Issue by Lockszmith** [zwave-js/zwavejs2mqtt#1226]
An issue (most likely with the controller)
My feature request is related to a problem:
I've been working with this USB stick for a while, going through different reincarnations of HA, and was really glad to take ZWaveJS for a spin.

I decided to hard-reset my Z-Wave controller and start the network from scratch, as it was a mess from previous manipulations.

(BTW, ZWave-JS is **AWESOME!** - fast, clear, easy to understand)

When adding a smoke detector ([First Alarm ZCOMBO](https://support.firstalert.com/s/active-product/a213l0000024tMuAAI)) with low battery, the inclusion started, introducing a node (node 22) but never completed.
After replacing the batteries, exclusion didn't remove the Node from the Controller, and to following inclusion introduced a new node (node 23).

Now I have an Unknown node with stuck in the ProtoclInfo state marked as Asleep, and nothing I do gets it out of this state.

![image](https://user-images.githubusercontent.com/905716/118381257-14f25300-b5b7-11eb-8b34-3603273e68ac.png)

I've tried Failed Nodes: Check, Re-Interview, Remove Associations - nothing changes the state of the node from Asleep to **failed** or **dead**.

**Describe the solution you'd like**
I recall in the Sigma windows application, there is a way to mark a node as dead, and then remove it.

I was hoping to find something like this in ZWaveJS2MQTT

**Describe alternatives you've considered**
I'm not sure what the overall negative impact is, other than an eye sore.
If this discussion provides more info, I might end up connecting the USB to a windows machine, clearing the dead node, and then hoping that when I plug it back to the HA server, everything will still work.

**Additional context**
[Z-Wave JS to MQTT Home Assistant Add-On](https://github.com/hassio-addons/addon-zwavejs2mqtt) [v0.15.0](https://github.com/hassio-addons/addon-zwavejs2mqtt/tree/v0.15.0)*
Z-Wave controller : [Nortek Control | HUSBZB-1 | GoControl QuickStick Combo](https://www.nortekcontrol.com/products/2gig/husbzb-1-gocontrol-quickstick-combo/) Purchased on [amazon](https://smile.amazon.com/gp/product/B01GJ826F8) August 2017.

*currently not integrated (on purpose) from home-assistant.

[Node #1 (Controller) Debug Info Json](https://h.lksz.me/fayeno.json)
[Node #22 (Broken) Debug Info Json](https://h.lksz.me/ezokak.json)
[Node #23 (Good) Debug Info Json](https://h.lksz.me/viyutu.json)

Let me know if you need me to collect anything else.

**Comment by robertsLando** [comment]
@Lockszmith Could you upload also logs from both z2m and zwavejs please?

**Comment by Lockszmith** [comment]
I knew I was missing something in my report: 🤦 
Here are the logs: https://lksz.me/zwjs-1226logs

I switched on Silley logs, stopped the service, cleared the log, and then started the service - so it should show you full initialization.

The node in question is Node 22.

I see the following in the logs in relation to node 22.
Part of the initialization of the controller:
```
2021-05-17T14:02:58.600Z CNTRLR   received node information:
                                    controller role:            primary
                                    controller is a SUC:        true
                                    controller is a slave:      false
                                    controller supports timers: false
                                    nodes in the network:       1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 1
                                  5, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 33
```
and then later:
> 2021-05-17T14:02:58.809Z CNTRLR   [Node 022] The node is asleep.
> 2021-05-17T14:02:58.810Z CNTRLR   [Node 022] Beginning interview - last completed stage: ProtocolInfo
> 2021-05-17T14:02:58.810Z CNTRLR » [Node 022] querying node info...

And afterwards as part of return-routes when querying other nodes, here is an example:
```
2021-05-17T14:02:58.833Z SERIAL « [ACK]                                                                   (0x06)
2021-05-17T14:02:58.834Z SERIAL « 0x012001803e0063070000000000000000000000000000000000000000000000000 (34 bytes)
                                  004
2021-05-17T14:02:58.834Z SERIAL » [ACK]                                                                   (0x06)
2021-05-17T14:02:58.835Z DRIVER « [RES] [GetRoutingInfo]
                                    node ids: 2, 3, 4, 5, 6, 17, 18, 22, 23, 25, 26, 27
2021-05-17T14:02:58.837Z SERIAL » 0x010700801100000069                                                 (9 bytes)
2021-05-17T14:02:58.837Z DRIVER » [REQ] [GetRoutingInfo]
                                    remove non-repeaters: false
                                    remove bad links:     false
2021-05-17T14:02:58.838Z CNTRLR « [Node 001] node neighbors received: 2, 3, 4, 5, 6, 17, 18, 22, 23, 25, 26, 27
```

I do see that the `GetRoutingInfo` has `remove non-repeaters` and `remove bad links` as `false`

I thought _Remove Association_ would do the trick, but I see in the logs that a call to Node 002 goes out, but nothing else.

```
2021-05-17T14:25:40.987Z SERIAL » 0x010b001302048e04001625134b                                        (13 bytes)
2021-05-17T14:25:40.987Z DRIVER » [Node 002] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      19
                                  └─[MultiChannelAssociationCCRemove]
                                      group id: undefined
                                      node ids: 22
2021-05-17T14:25:40.990Z SERIAL « [ACK]                                                                   (0x06)
2021-05-17T14:25:40.994Z SERIAL « 0x0104011301e8                                                       (6 bytes)
2021-05-17T14:25:40.994Z SERIAL » [ACK]                                                                   (0x06)
2021-05-17T14:25:40.994Z DRIVER « [RES] [SendData]
                                    was sent: true
```

# Bottom Line
While I don't think the issue is any way with ZWave-JS, it's probably more to do with the controller, I do believe exposing some additional functionality (forcibly mark a node as dead) would help tidy things up.

**Comment by robertsLando** [comment]
To mark a node as failed you should use `removeFailedNode` api. Check docs https://zwave-js.github.io/node-zwave-js/#/api/controller?id=removefailednode

**Comment by Lockszmith** [comment]
@robertsLando, the way I understand it - `removeFailedNode` will only remove a node the controller has marked `failed`, but in my case, the node is never marked that way.
